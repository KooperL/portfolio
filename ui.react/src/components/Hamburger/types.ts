export interface Props {
  data: {
    label: string;
    destination: string;
    callback?: Function
  }[]
}

