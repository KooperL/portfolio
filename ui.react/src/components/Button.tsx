import { Scheme } from "../containers/context/type";

export function Button(props: {colours: Scheme, disabled?: Boolean}) {
  const style = {
    backgroundColor: props.colours.button.bgSolid,
    color: props.colours.button.text,
    borderRadius: '10px',
    padding: '0.2rem 2rem',
    // margin: '10px'
  } as const;
  return (
    <button className="submit-button" type='submit' name='submit' value='Submit' style={style}>Submit</button>
  );
}

// https://stackoverflow.com/questions/28365233/inline-css-styles-in-react-how-to-implement-ahover