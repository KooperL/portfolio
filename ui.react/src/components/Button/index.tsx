import { Scheme } from "../../containers/context/type"

export function Button(props: {
  colours: Scheme
  disabled?: Boolean
  callBack?: () => void
  label?: string
  action?: "button" | "submit" | "reset" | undefined
}) {
  const style = {
    backgroundColor: props.colours.button.bgSolid,
    color: props.colours.button.text,
    borderRadius: "10px",
    padding: "0.2rem 2rem",
    // margin: '10px'
  } as const
  if (props.callBack) {
    return (
      <div className="submit-button-container">
        {/* @ts-ignore - Sooo bad.... cmon ts, it exists!!!!  */}
        <button
          className="submit-button"
          value={props.label ?? "submit"}
          style={style}
          onClick={() => {
            props.callBack && props.callBack()
          }}
        >
          {props.label ?? "Submit"}
        </button>
      </div>
    )
  }
  return (
    <button
      className="submit-button"
      type={props.action ?? "submit"}
      name="submit"
      value={props.label ?? "Submit"}
      style={style}
    >
      {props.label ?? "Submit"}
    </button>
  )
}

// https://stackoverflow.com/questions/28365233/inline-css-styles-in-react-how-to-implement-ahover
