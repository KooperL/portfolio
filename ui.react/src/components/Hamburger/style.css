.hamburger {
}

.hamburger .icon {
  width: 5rem;
  height: 5rem;
  color: black;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  border-radius: 7px;
  border-top-left-radius: 0;
}

.hamburger .hamburger-icon {
  font-size: xx-large;
}

.hamburger .hover-effect {
  position: absolute;
  background-color: rgba(255, 255, 255, 0);
  width: 90%;
  height: 90%;
  border-radius: 10px;
  border-top-left-radius: 0;
  transition: 0.2s;
}

.hamburger .hover-effect:hover {
  /* transform: scale(1.5); */
  background-color: rgba(255, 255, 255, 0.30);
}



.hamburger .links {
  position: absolute;
  top: 100%;
  left: 0;
  border-radius: 0.5rem;
  height: 0;
  transition: height 1.5s ease-in-out;
}

.links.open {
  height: fit-content;
}

@media screen and (max-width: 1000px) {
  .hamburger .links {
    width: 100%;
    padding-bottom: 1rem;
  }
}
