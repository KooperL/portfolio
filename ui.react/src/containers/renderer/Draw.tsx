import { triangle } from "./utils/types";

export function DrawFigure(props: {name: string, h: number | string, w: number | string}) {
  // const displayGCD = gcd(+props.w, +props.h);
  return (
    <figure>
      {/* <svg viewBox={`${0-+props.w/2} ${0-+props.h/2} ${+props.w/2} ${+props.h/2}`} width={props.w} height={props.h}> */}
      <svg viewBox={`0 0 ${+props.w} ${+props.h}`} width={props.w} height={props.h}>
        <use xlinkHref={`#${props.name}`}/>
      </svg>
    </figure>
  );
}

export function drawBlueprint(props: {name: string, tris: triangle[]}) {
  return (
    <svg className="defs-only" xmlns="http://www.w3.org/2000/svg">
      <symbol id={props.name}>
        {props.tris.map((triangle, triangleIndex) => (
          <polygon key={triangleIndex}
            points={`${triangle.p[0].x},${triangle.p[0].y} ${triangle.p[1].x},${triangle.p[1].y} ${triangle.p[2].x},${triangle.p[2].y}`}
            style={{
              fill: triangle['c']['fill'] ?? 'red',
              stroke: triangle['c']['stroke'] ?? 'pink',
              strokeWidth: triangle['c']['strokeWidth'] || 5.0,
            }}
          />
          ))}
      </symbol>
    </svg>
  );
}