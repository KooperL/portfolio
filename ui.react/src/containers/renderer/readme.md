Pressing pause for now.
 TODO:
 - Vector transformation as a method of their class. This will let them keep their colour properties when undergoing a transformation
 - Edge plane clipping for simple raster
 - Try to do something to reduce duplicate code
 - Use ticker as switch to toggle different transformations for grid
 - Do the scale properly -> make it a method and take args
 - colour spectrum