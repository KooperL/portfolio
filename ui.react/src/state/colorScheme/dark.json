{
  "header": {
    "background": "#205375",
    "text": "#EFEFEF",
    "outline": "#EFEFEF"
  },
  "body": {
    "background": "#112B3C",
    "foreground": "#205375",
    "h1": "#F66B0E",
    "h2": "#EFEFEF",
    "text": "#F2F2F2"
  },
  "button": {
    "bgSolid": "#205375",
    "bgGrad1": "#455375",
    "bgGrad2": "#20535a",
    "text": "#F2F2F2"
  }
}
