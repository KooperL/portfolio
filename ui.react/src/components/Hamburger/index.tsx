import { useContext, useEffect, useState } from "react"
import "./style.css"
import { SchemeContext } from "../../containers/context/colourScheme"
import { Props } from "./types"
import { Link } from "react-router-dom"
import { postMonitor } from "../../containers/App/api/loggerApi"
import { LoggingPOSTResponse } from "../Logger/types"
import ButtonRedir from "../ButtonRedir"

function Hamburger(props: Props) {
  const [scheme, setScheme] = useContext(SchemeContext)
  const [open, setOpen] = useState(false)

  useEffect(() => {}, [open])

  const burgerButton = (
    <>
      <button
        onClick={() => {
          setOpen(!open)
        }}
      >
        <div
          className="icon"
          style={{ backgroundColor: scheme.body.h1 }}
        >
          <p className="hamburger-icon">{open ? "×" : "≡"}</p>
          <div className="hover-effect"></div>
        </div>
      </button>
    </>
  )

  // if(!open) {
  //   return (
  //     <div className="hamburger">
  //       {burgerButton}
  //       <div className={`links ${open ? 'open' : ''}`} style={{backgroundColor: scheme.body.background}}></div>
  //     </div>
  //   )
  // }

  return (
    <div className="hamburger">
      {burgerButton}
      <div
        className={`links ${open ? "open" : ""}`}
        style={{ backgroundColor: scheme.body.background }}
      >
        {/* <div className="links" style={{display: open ? 'fixed' : 'none'}}> */}
        {open ? (
          props.data.map((item, index) => (
            <ButtonRedir
              destination={item.destination}
              label={item.label}
              key={index}
              local={true}
              onClickCallback={() => {
                setOpen(false)
                if (item.callback) item.callback()
              }}
            />
          ))
        ) : (
          <></>
        )}
      </div>
    </div>
  )
}

export default Hamburger
