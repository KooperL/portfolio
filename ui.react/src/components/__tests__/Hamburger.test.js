import { Button } from "../Button"
import React from "react"
import renderer from "react-test-renderer"
import dark from "../../state/colorScheme/dark.json"
import Hamburger from "../Hamburger"

describe("Hamburger component", () => {
  test("it matches the snapshot", () => {
    const tree = renderer
      .create(
        <Hamburger
          data={[
            {
              destination: "/",
              label: "test1",
              callback: () => {},
            },
            {
              destination: "/",
              label: "test2",
              callback: () => {},
            },
          ]}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
