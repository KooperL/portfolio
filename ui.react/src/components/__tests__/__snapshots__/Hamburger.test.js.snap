// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger component it matches the snapshot 1`] = `
<div
  className="hamburger"
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#205375",
      }
    }
  >
    <div
      className="icon"
      style={
        Object {
          "backgroundColor": "#F66B0E",
        }
      }
    >
      <p
        className="hamburger-icon"
      >
        â‰¡
      </p>
      <div
        className="hover-effect"
      />
    </div>
  </button>
  <div
    className="links "
    style={
      Object {
        "backgroundColor": "#112B3C",
      }
    }
  />
</div>
`;
